core/src/main/scala/org/apache/spark/scheduler/ReplayListenerBus.scala

```diff
@@ -43,19 +43,25 @@ private[spark] class ReplayListenerBus extends SparkListenerBus with Logging {
    * @param sourceName Filename (or other source identifier) from whence @logData is being read
    * @param maybeTruncated Indicate whether log file might be truncated (some abnormal situations
    *        encountered, log file might not finished writing) or not
+   * @param eventsFilter Filter function to select JSON event strings in the log data stream that
+   *        should be parsed and replayed. When not specified, all event strings in the log data
+   *        are parsed and replayed.
    */
   def replay(
       logData: InputStream,
       sourceName: String,
-      maybeTruncated: Boolean = false): Unit = {
+      maybeTruncated: Boolean = false,
+      eventsFilter: (String) => Boolean = (s: String) => true): Unit = {
     var currentLine: String = null
     var lineNumber: Int = 1
     try {
       val lines = Source.fromInputStream(logData).getLines()
```

Comment:
I don't think that logic changes. It already doesn't actually know if there are more lines in the file, because the error might cause the iterator to close and not indicate any more lines available. It's pretty best-effort. I guess the scenario is: error occurs, and there are no more matching lines, but there are some unmatching lines available? Now we'd continue instead of fail, which seems equally reasonable given the purpose.